### PR TITLE
feat: change the way throw error

### PR DESCRIPTION
**Pull request by mapleeit**
Don't throw error during new Client, if there
is no authentication info. Instead, throw error when
actually calling according API.

Related to: https://github.com/jina-ai/auth/issues/2

---

@bwanglzu This is a breaking change, please change the usage accordingly in finetuner.

### CODE REVIEW

File: .github/workflows/style-and-test-check.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
       - name: Run tests
         run: make test
         env:
-          HUBBLE_ACCESS_TOKEN: ${{secrets.HUBBLE_ACCESS_TOKEN}}
+          JINA_AUTH_TOKEN: ${{secrets.HUBBLE_ACCESS_TOKEN}}
```
Comment (bwanglzu) [inline .github/workflows/style-and-test-check.yml:22]: how about change the secrets as well? here and in github secrets, and maybt CI

File: README.md [hunk 2]
```diff
@@ -109,11 +109,36 @@ response = client.list_artifacts(filter={'metaData.foo': 'bar'}, sort={'type': -
...
+
+client = hubble.Client()
+
+try:
+    client.get_user_info()
+except Exception as ex:
+    if (isinstance(ex, hubble.excepts.AuthenticationRequiredError)):
+        print('Please login first.')
+    else:
+        print('Unknown error')
```
Comment (bwanglzu) [inline README.md:124]: i would write it like this:

```python
except hubble.excepts.AuthenticationRequiredError:
    # do something
except exception
    # do something else
```

File: tests/unit/client/test_client.py [hunk 3]
```diff
@@ -1,18 +1,16 @@
...
 from hubble import Client
 from hubble.excepts import AuthenticationRequiredError
 
 
-def test_handle_error_request(mocker):
-    mocker.patch('hubble.utils.auth.Auth.get_auth_token', return_value='fake-token')
-    with pytest.raises(AuthenticationRequiredError):
-        client = Client()
-        client.get_user_info()
+@patch.dict(os.environ, {"JINA_AUTH_TOKEN": ""})
```
Comment (bwanglzu) [inline tests/unit/client/test_client.py:9]: single quote

**Review by bwanglzu — CHANGES_REQUESTED** [review]
minor comments

**Comment by mapleeit** [comment]
Comments addressed.

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Pull request merged**